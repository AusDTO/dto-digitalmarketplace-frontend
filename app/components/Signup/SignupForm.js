import React from 'react'
import { Form } from 'react-redux-form'

import { required, validEmail, governmentEmail } from '../../components/validators'

import Layout from '../../components/shared/Layout'
import ErrorBox from '../../components/shared/form/ErrorBox'
import Textfield from '../../components/shared/form/Textfield'
import RadioList from '../../components/shared/form/RadioList'
import RadioListBox from '../../components/shared/form/RadioListBox/RadioListBox'
import PageAlert from '@gov.au/page-alerts'

export const SignupForm = ({
  csrf_token,
  model,
  signupForm,
  buyer_url,
  seller_url,
  signupSuccess,
  submitClicked,
  handleSubmit,
  submitFailed,
  isBuyer,
  emailValidators,
  emailErrorMessages
}) => {
  let employmentStatus = signupForm.employment_status
  let action = isBuyer ? buyer_url : seller_url

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <div className="row">
      <div className="col-sm-push-2 col-sm-8 col-xs-12">
        <div>
          {signupSuccess &&
            <div>
              <PageAlert as="success">
                <h4>Signup email sent</h4>
              </PageAlert>
              <article role="main">
                <header className="page-heading page-heading-without-breadcrumb">
                  <h1>Thanks for requesting access to the Digital Marketplace.</h1>
                </header>
                {isBuyer && signupForm.employment_status === 'contractor'
                  ? <div>
                      <p>
                        An email has been sent to your manager at <strong>{signupForm.line_manager_email}</strong> with
                        next steps.
                      </p>
                      <p>
                        If they don’t receive the email within the next 5 minutes or so, check to see if it’s been
                        classified as spam or
                        <a href="/contact-us" target="_blank" rel="external">
                          {' '}contact us{' '}
                        </a>{' '}
                        for assistance.
                      </p>
                    </div>
                  : <div>
                      <p>
                        An email has been sent to <strong>{signupForm.email_address}</strong> with next steps.
                      </p>
                      <p>
                        If you don’t receive the email within the next 5 minutes or so, check to see if it’s been
                        classified as spam or
                        <a href="/contact-us" target="_blank" rel="external">
                          {' '}contact us{' '}
                        </a>{' '}
                        for assistance.
                      </p>
                    </div>}
              </article>
            </div>}
          {!signupSuccess &&
            <Layout>
              <ErrorBox
                title="There was a problem with signup"
                model={model}
                submitClicked={submitClicked}
                setFocus={setFocus}
              />
              <header>
                <h1>Let’s get started</h1>
              </header>
              <article role="main">
                <Form
                  model={model}
                  action={action}
                  method="post"
                  id="signup"
                  onSubmit={model => handleSubmit(model)}
                  onSubmitFailed={submitFailed}
                >
                  {csrf_token && <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />}
                  <div className="user-type">
                    <RadioListBox
                      model={`${model}.user_type`}
                      name="user_type"
                      id="user_type"
                      label="Choose the option that matches your situation."
                      options={[
                        {
                          value: 'buyer',
                          label: (
                            <span>
                              <span>Buyer</span>
                              <p>I want to buy on behalf of</p>
                              <p>government.</p>
                            </span>
                          )
                        },
                        {
                          value: 'seller',
                          label: (
                            <span>
                              <span>Seller</span>
                              <p>I want to sell digital products or</p>
                              <p>services.</p>
                            </span>
                          )
                        }
                      ]}
                      validators={{
                        required
                      }}
                      messages={{
                        required: 'You must select a user type'
                      }}
                    />
                  </div>
                  Now enter your name and your work email address.
                  <Textfield
                    model={`${model}.name`}
                    name="name"
                    id="name"
                    htmlFor="name"
                    label="Full name"
                    validators={{
                      required
                    }}
                    messages={{
                      required: 'Your name is required'
                    }}
                  />
                  <Textfield
                    model={`${model}.email_address`}
                    name="email_address"
                    id="email_address"
                    type="email"
                    htmlFor="email_address"
                    label={
                      isBuyer
                        ? <span>
                            Email address ending in
                            <b>.gov.au.</b>
                          </span>
                        : 'Email address'
                    }
                    description={
                      isBuyer
                        ? <span>
                            If your email is different, request your account from{' '}
                            <a href="mailto:marketplace@digital.gov.au">marketplace@digital.gov.au</a>.
                          </span>
                        : ''
                    }
                    validators={emailValidators}
                    messages={emailErrorMessages}
                  />
                  {isBuyer &&
                    <div className="employment-status">
                      <RadioList
                        model={`${model}.employment_status`}
                        name="employment_status"
                        id="employment_status"
                        label="To create a buyer account you need to be a government employee or be authorised by a government employee. Choose the option below that matches your situation."
                        options={[
                          {
                            value: 'employee',
                            label:
                              'I am an employee under the Commonwealth Public Service Act (1999) or under equivalent state or territory legislation and need access to the Digital Marketplace to perform my role.'
                          },
                          {
                            value: 'contractor',
                            label: 'I am a contractor working in local, state or federal government.'
                          }
                        ]}
                        validators={{
                          required
                        }}
                        messages={{
                          required: 'You must specify your employment status.'
                        }}
                      />
                    </div>}
                  {employmentStatus &&
                    employmentStatus === 'contractor' &&
                    isBuyer &&
                    <div>
                      <p>To create your account we will also need approval from a line manager who:</p>
                      <ul>
                        <li>
                          Is an employee under the Commonwealth Public Service Act (1999) or under equivalent State or
                          Territory legislation, and
                        </li>
                        <li>is satisfied you need to access the Digital Marketplace.</li>
                      </ul>
                      <Textfield
                        model={`${model}.line_manager_name`}
                        name="line_manager_name"
                        id="line_manager_name"
                        htmlFor="line_manager_name"
                        label="Your manager's full name"
                        validators={{
                          required
                        }}
                        messages={{
                          required: 'You must provide the name of your manager'
                        }}
                      />
                      <Textfield
                        model={`${model}.line_manager_email`}
                        name="line_manager_email"
                        id="line_manager_email"
                        htmlFor="line_manager_email"
                        label="Your manager's email address"
                        validators={{
                          required,
                          validEmail,
                          governmentEmail
                        }}
                        messages={{
                          required: "You must provide your manager's email address",
                          validEmail: 'A validly formatted email is required.',
                          governmentEmail: ' Email should have a government domain.'
                        }}
                      />
                      <PageAlert as="info">
                        <p>
                          Remember to let this person know we’ll be sending them an email requesting their
                          authorisation.
                        </p>
                      </PageAlert>
                    </div>}
                  <p>
                    <small>
                      By creating an account you confirm your acceptance of our{' '}
                      <a href="/terms-of-use" target="_blank" rel="external">
                        Terms of Use
                      </a>
                    </small>
                  </p>
                  <input className="uikit-btn" type="submit" value="Create your account" onClick={submitClicked} />
                </Form>
              </article>
            </Layout>}
        </div>
      </div>
    </div>
  )
}
