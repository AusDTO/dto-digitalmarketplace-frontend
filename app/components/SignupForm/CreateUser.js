import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { decodeInviteToken } from '../../actions/memberActions'

class CreateUser extends Component {
  componentDidMount() {
    let tokenString = this.props.match.params.tokenstring
    this.props.loadRegistrationData(tokenString)
  }

  shouldComponentUpdate(nextProps) {
    if (this.props.registrationDataLoaded !== nextProps.registrationDataLoaded) {
      return true
    }
    return false
  }

  render() {
    return (
      <div>
        {this.props.registrationDataLoaded ? this.props.userRegisterDetails.name : null}
      </div>
    )
  }
}

CreateUser.propTypes = {
  userRegisterDetails: PropTypes.shape({
    name: PropTypes.string.isRequired,
    email_address: PropTypes.string.isRequired,
    manager_name: PropTypes.string,
    manager_email: PropTypes.string
  }),
  hasErrored: PropTypes.bool,
  isLoading: PropTypes.bool,
  registrationDataLoaded: PropTypes.bool,
  loadRegistrationData: PropTypes.func.isRequired
}

const mapStateToProps = ({ createUser }) => {
  return {
    userRegisterDetails: createUser.userRegisterDetails,
    hasErrored: createUser.memberInfoHasErrored,
    isLoading: createUser.memberInfoIsLoading,
    registrationDataLoaded: createUser.registrationDataLoaded
  }
}

const mapDispatchToProps = dispatch => {
  return {
    loadRegistrationData: tokenString => {
      dispatch(decodeInviteToken(tokenString))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CreateUser)
