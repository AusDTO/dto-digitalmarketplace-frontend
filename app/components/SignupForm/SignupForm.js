import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Form } from 'react-redux-form'

import { required, validEmail, governmentEmail } from '../../components/validators'

import Layout from '../../components/shared/Layout'
import BaseForm from '../../components/shared/form/BaseForm'
import ErrorBox from '../../components/shared/form/ErrorBox'
import Textfield from '../../components/shared/form/Textfield'
import formProps from '../../components/shared/form/formPropsSelector'
import RadioList from '../../components/shared/form/RadioList'
import RadioListBox from '../../components/shared/form/RadioListBox/RadioListBox'
import PageAlert from '@gov.au/page-alerts'

import { handleSignupSubmit } from '../../actions/memberActions'

class SignupForm extends BaseForm {
  static propTypes = {
    action: PropTypes.string,
    csrf_token: PropTypes.string,
    form: PropTypes.object.isRequired
  }

  constructor(props) {
    super(props)
    this.state = {
      signupSuccess: null,
      emailValidators: {
        required,
        validEmail
      },
      emailErrorMessages: {
        required: 'Your email is required',
        validEmail: 'A validly formatted email is required.',
        governmentEmail: ' Email should have a government domain.'
      },
      isBuyer: this.props.signupForm.user_type === 'buyer',
      submitClicked: null
    }
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.form.valid) {
      if (!nextProps.form.valid) {
        this.setState({
          signupSuccess: null
        })
      }
    }

    if (this.props.signupForm.user_type !== nextProps.signupForm.user_type) {
      if (nextProps.signupForm.user_type === 'seller') {
        this.setState({
          emailValidators: {
            required,
            validEmail
          },
          emailErrorMessages: {
            required: 'Your email is required',
            validEmail: 'A validly formatted email is required.'
          },
          isBuyer: false
        })
      }

      if (nextProps.signupForm.user_type === 'buyer') {
        this.setState({
          emailValidators: {
            required,
            validEmail,
            governmentEmail
          },
          emailErrorMessages: {
            required: 'Your email is required',
            validEmail: 'A validly formatted email is required.',
            governmentEmail: ' Email should have a government domain.'
          },
          isBuyer: true
        })
      }
    }

    if (this.props.signupSuccess !== nextProps.signupSuccess) {
      this.setState({
        signupSuccess: nextProps.signupSuccess
      })
    }
  }

  handleSubmit(model) {
    this.setState({ signupSuccess: null })
    this.props.handleSignupSubmit(model)
  }

  onSubmitFailed() {
    window.scrollTo(0, 0)
  }

  onSubmitClicked = () => {
    this.setState({
      submitClicked: new Date().valueOf()
    })
  }

  render() {
    const { csrf_token, model, children, signupForm, buyer_url, seller_url } = this.props
    let employmentStatus = signupForm.employment_status
    let action = isBuyer ? buyer_url : seller_url
    let { signupSuccess, isBuyer, emailValidators, emailErrorMessages } = this.state

    let hasFocused = false
    const setFocus = e => {
      if (!hasFocused) {
        hasFocused = true
        e.focus()
      }
    }

    return (
      <div className="row">
        <div className="col-sm-push-2 col-sm-8 col-xs-12">
          <div>
            {signupSuccess &&
              <div>
                <PageAlert as="success">
                  <h4>Signup email sent</h4>
                </PageAlert>
                <article role="main">
                  <header className="page-heading page-heading-without-breadcrumb">
                    <h1>Thanks for requesting access to the Digital Marketplace.</h1>
                  </header>
                  {isBuyer && signupForm.employment_status === 'contractor'
                    ? <div>
                        <p>
                          An email has been sent to your manager at <strong>{signupForm.line_manager_email}</strong>{' '}
                          with next steps.
                        </p>
                        <p>
                          If they don’t receive the email within the next 5 minutes or so, check to see if it’s been
                          classified as spam or
                          <a href="/contact-us" target="_blank" rel="external">
                            {' '}contact us{' '}
                          </a>{' '}
                          for assistance.
                        </p>
                      </div>
                    : <div>
                        <p>
                          An email has been sent to <strong>{signupForm.email_address}</strong> with next steps.
                        </p>
                        <p>
                          If you don’t receive the email within the next 5 minutes or so, check to see if it’s been
                          classified as spam or
                          <a href="/contact-us" target="_blank" rel="external">
                            {' '}contact us{' '}
                          </a>{' '}
                          for assistance.
                        </p>
                      </div>}
                </article>
              </div>}
            {!signupSuccess &&
              <Layout>
                <header>
                  <h1>Let’s get started</h1>
                </header>
                <article role="main">
                  <ErrorBox
                    title="There was a problem with signup"
                    model={model}
                    submitClicked={this.state.submitClicked}
                    setFocus={setFocus}
                  />
                  <Form
                    model={model}
                    action={action}
                    method="post"
                    id="signup"
                    onSubmit={model => this.handleSubmit(model)}
                    onSubmitFailed={this.onSubmitFailed}
                  >
                    {csrf_token && <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />}
                    <div className="user-type">
                      <RadioListBox
                        model={`${model}.user_type`}
                        name="user_type"
                        id="user_type"
                        label="Choose the option that matches your situation."
                        options={[
                          {
                            value: 'buyer',
                            label: (
                              <span>
                                <span>Buyer</span>
                                <p>I want to buy on behalf of</p>
                                <p>government.</p>
                              </span>
                            )
                          },
                          {
                            value: 'seller',
                            label: (
                              <span>
                                <span>Seller</span>
                                <p>I want to sell digital products or</p>
                                <p>services.</p>
                              </span>
                            )
                          }
                        ]}
                        validators={{
                          required
                        }}
                        messages={{
                          required: 'You must select a user type'
                        }}
                      />
                    </div>
                    Now enter your name and your work email address.
                    <Textfield
                      model={`${model}.name`}
                      name="name"
                      id="name"
                      htmlFor="name"
                      label="Full name"
                      validators={{
                        required
                      }}
                      messages={{
                        required: 'Your name is required'
                      }}
                    />
                    <Textfield
                      model={`${model}.email_address`}
                      name="email_address"
                      id="email_address"
                      type="email"
                      htmlFor="email_address"
                      label={
                        isBuyer
                          ? <span>
                              Email address ending in
                              <b>.gov.au.</b>
                            </span>
                          : 'Email address'
                      }
                      description={
                        isBuyer
                          ? <span>
                              If your email is different, request your account from{' '}
                              <a href="mailto:marketplace@digital.gov.au">marketplace@digital.gov.au</a>.
                            </span>
                          : ''
                      }
                      validators={emailValidators}
                      messages={emailErrorMessages}
                    />
                    {isBuyer &&
                      <div className="employment-status">
                        <RadioList
                          model={`${model}.employment_status`}
                          name="employment_status"
                          id="employment_status"
                          label="To create a buyer account you need to be a government employee or be authorised by a government employee. Choose the option below that matches your situation."
                          options={[
                            {
                              value: 'employee',
                              label:
                                'I am an employee under the Commonwealth Public Service Act (1999) or under equivalent state or territory legislation and need access to the Digital Marketplace to perform my role.'
                            },
                            {
                              value: 'contractor',
                              label: 'I am a contractor working in local, state or federal government.'
                            }
                          ]}
                          validators={{
                            required
                          }}
                          messages={{
                            required: 'You must specify your employment status.'
                          }}
                        />
                      </div>}
                    {employmentStatus &&
                      employmentStatus === 'contractor' &&
                      isBuyer &&
                      <div>
                        <p>To create your account we will also need approval from a line manager who:</p>
                        <ul>
                          <li>
                            Is an employee under the Commonwealth Public Service Act (1999) or under equivalent State or
                            Territory legislation, and
                          </li>
                          <li>is satisfied you need to access the Digital Marketplace.</li>
                        </ul>
                        <Textfield
                          model={`${model}.line_manager_name`}
                          name="line_manager_name"
                          id="line_manager_name"
                          htmlFor="line_manager_name"
                          label="Your manager's full name"
                          validators={{
                            required
                          }}
                          messages={{
                            required: 'You must provide the name of your manager'
                          }}
                        />
                        <Textfield
                          model={`${model}.line_manager_email`}
                          name="line_manager_email"
                          id="line_manager_email"
                          htmlFor="line_manager_email"
                          label="Your manager's email address"
                          validators={{
                            required,
                            validEmail,
                            governmentEmail
                          }}
                          messages={{
                            required: "You must provide your manager's email address",
                            validEmail: 'A validly formatted email is required.',
                            governmentEmail: ' Email should have a government domain.'
                          }}
                        />
                        <PageAlert as="info">
                          <p>
                            Remember to let this person know we’ll be sending them an email requesting their
                            authorisation.
                          </p>
                        </PageAlert>
                      </div>}
                    {children}
                    <p>
                      <small>
                        By creating an account you confirm your acceptance of our{' '}
                        <a href="/terms-of-use" target="_blank" rel="external">
                          Terms of Use
                        </a>
                      </small>
                    </p>
                    <input
                      className="uikit-btn"
                      type="submit"
                      value="Create your account"
                      onClick={this.onSubmitClicked}
                    />
                  </Form>
                </article>
              </Layout>}
          </div>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => {
  return {
    handleSignupSubmit: model => dispatch(handleSignupSubmit(model))
  }
}

const mapStateToProps = state => {
  return {
    ...formProps(state, 'signupForm'),
    signupSuccess: state.user.signupSuccess,
    isDuplicate: state.user.isDuplicate
  }
}

export { Textfield, mapStateToProps, SignupForm as Form }

export default connect(mapStateToProps, mapDispatchToProps)(SignupForm)
