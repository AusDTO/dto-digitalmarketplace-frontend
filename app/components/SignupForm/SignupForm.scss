.user-type .radio-list-container{
	display: block;
  width: 100%;
	border: 2px solid #bbbbbb;
	background-color: #e8f5fa;
  margin: 0;
  padding: 15px 12px 0px 20px;
}

.employment-status .uikit-control-input--full {
  margin-bottom: 0.4em;
  margin-top: 0.4em;
}

.user-type .uikit-control-input__input:focus+.uikit-control-input__text {
  outline: none;
}

.user-type .uikit-control-input__input:focus+.uikit-control-input__text > span {
  outline: 1px solid #313131;
}

@media all and (max-width: 1024px) {
  .user-type .radio-list-container:first-child {
    margin-bottom: 20px;
  }
}

@media all and (min-width: 1024px)  {
  .user-type .radio-list-container {
    width: 48%;
    display:inline-block;
  }

  .user-type .radio-list-container:first-child {
    margin-right: 20px;
  }
}

.user-type fieldset span input[type='radio'] + label > p {
  font-size: .875rem;
  font-weight: 400;
  margin-bottom: 0.1em;
  padding-left: 35px;
}