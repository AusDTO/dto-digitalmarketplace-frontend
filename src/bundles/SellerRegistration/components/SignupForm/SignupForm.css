.user-type fieldset span input[type='radio'] + label{
	display:inline-block;
	width:50%;
	border: 2px solid #bbbbbb;
}

.user-type fieldset span input[type='radio']:checked + label{
	outline:none;
	border: 2px solid #717171;
}

.employment-status div fieldset legend {
	font-weight: normal;
}