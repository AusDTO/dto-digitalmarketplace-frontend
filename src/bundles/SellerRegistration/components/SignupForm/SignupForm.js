import React from 'react';
import { connect } from 'react-redux';
import { Form } from 'react-redux-form';

import { required } from '../../../../validators';

import Layout        from '../../../../shared/Layout';
import BaseForm      from '../../../../shared/form/BaseForm';
import SubmitForm    from '../../../../shared/form/SubmitForm';
import ErrorBox      from '../../../../shared/form/ErrorBox';
import Textfield     from '../../../../shared/form/Textfield';
import formProps     from '../../../../shared/reduxModules/formPropsSelector';

class SignupForm extends BaseForm {

  static propTypes = {
    action: React.PropTypes.string,
    csrf_token: React.PropTypes.string,
    form: React.PropTypes.object.isRequired
  }

  render() {
    const { action, csrf_token, model, form, children, onSubmit } = this.props;
    return (
      <Layout>
        <header>
          <h1>Claim your spot in the Marketplace</h1>
          <p>
            Weâ€™re looking for the very best digital businesses to help deliver 21st century digital products and services to government through the Digital Marketplace.
          </p>
          <p>
            And the rewards? Only the chance to share in the $5.6B spent each year on ICT.
          </p>
          <h2>Ready to get started?</h2>
        </header>
        <article role="main">
          <ErrorBox focusOnMount={true} model={model}/>
          <Form model={model}
            action={action}
            method="post"
            id="signup"
            component={SubmitForm}
            valid={form.valid}
            onCustomSubmit={onSubmit}
          >
            {csrf_token && (
              <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />
            )}

            <Textfield
              model={`${model}.name`}
              name="name"
              id="name"
              htmlFor="name"
              label="Your full name"
              validators={{ required }}
              messages={{
                required: 'Name is required',
              }}
            />

            <Textfield
              model={`${model}.email`}
              name="email"
              id="email"
              htmlFor="email"
              label="Email address"
              validators={{ required }}
              messages={{
                required: 'Email is required',
              }}
            />

            {children}

            <input type="submit" value='Create seller account' role="button" />
            <p>
              <small>
                By requesting an account invitation you agree to the <a href="/terms-of-use">Terms of Use</a>.
              </small>
            </p>             
          </Form>
        </article>
      </Layout>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    ...formProps(state, 'signupForm')
  }
}

export {
  Textfield,
  mapStateToProps,
  SignupForm as Form
}

export default connect(mapStateToProps)(SignupForm);
