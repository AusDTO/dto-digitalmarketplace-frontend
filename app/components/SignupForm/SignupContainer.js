import React from 'react'
import { withRouter, Switch, Route } from 'react-router-dom'
import SignupForm from './SignupForm'
import UserOnboardingContainer from '../Onboarding/OnboardingContainer'
import CreateUserPage from '../../pages/CreateUserPage'
import NotFound from '../shared/NotFound'

const Routes = ({ match }) =>
  <Switch>
    <Route exact path={match.url} component={SignupForm} />
    <Route path={`${match.url}/createuser/:token`} component={CreateUserPage} />
    <Route path={`${match.url}/success`} component={UserOnboardingContainer} />
    <Route component={NotFound} />
  </Switch>

export default withRouter(Routes)
