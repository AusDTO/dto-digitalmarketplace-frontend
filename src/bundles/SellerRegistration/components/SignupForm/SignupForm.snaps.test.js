// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');
//jest.mock('react-router');

import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router';
import renderer from 'react-test-renderer';

import SignupForm, { Textfield } from './SignupForm';
import createStore from '../../redux/create';

test('SignupForm renders', () => {
  let store = createStore(Object.assign({}, { _serverContext: {} }))
  const component = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <SignupForm />
      </Provider>
    </BrowserRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('SignupForm renders with form_options', () => {
  const form_options = {
    csrf_token: 'sometoken',
    action: '/foo/bar'
  }
  let store = createStore(Object.assign({}, { _serverContext: {}, form_options }))
  const component = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <SignupForm />
      </Provider>
    </BrowserRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
