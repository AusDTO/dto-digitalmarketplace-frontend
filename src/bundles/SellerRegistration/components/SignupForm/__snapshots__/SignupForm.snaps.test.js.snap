// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm renders 1`] = `
<div>
  <div
    className="layout"
  >
    <header>
      <h1
        className="au-display-xl"
      >
        Let’s get started
      </h1>
    </header>
    <article
      role="main"
    >
      <form
        id="signup"
        method="post"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          styleName="user-type"
        >
          <div
            className="field"
          >
            <fieldset>
              <legend>
                Have you got digital expertise to sell to government? Or do you want to buy digital products and services on behalf of your government office? Choose the option that matches your situation.
              </legend>
              <div>
                <span>
                  <input
                    checked={false}
                    disabled={false}
                    id="user_type-buyer"
                    name="user_type"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    type="radio"
                    value="buyer"
                  />
                  <label
                    htmlFor="user_type-buyer"
                  >
                    <span>
                      Buyer
                      <p>
                        I want to buy on behalf of
                        <br />
                        government.
                      </p>
                    </span>
                  </label>
                </span>
                <span>
                  <input
                    checked={false}
                    disabled={false}
                    id="user_type-seller"
                    name="user_type"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    type="radio"
                    value="seller"
                  />
                  <label
                    htmlFor="user_type-seller"
                  >
                    <span>
                      Seller
                      <p>
                        I want to sell digital products or
                        <br />
                        services.
                      </p>
                    </span>
                  </label>
                </span>
              </div>
            </fieldset>
          </div>
        </div>
        Now enter your name and your work email address.
        <div
          className="field"
        >
          <label
            className="question-heading"
            htmlFor="name"
          >
            Full name
          </label>
          <input
            className=""
            disabled={false}
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
          />
        </div>
        <div
          className="field"
        >
          <label
            className="question-heading"
            htmlFor="email_address"
          >
            Email address
          </label>
          <p
            className="hint"
            id="email_address-hint"
          >
            <br />
          </p>
          <input
            className=""
            disabled={false}
            id="email_address"
            name="email_address"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="email"
          />
        </div>
        <p>
          <small>
            By creating an account you confirm your acceptance of our
            <a
              href="/terms-of-use"
              rel="noopener noreferrer"
              target="_blank"
            >
               Terms of Use
            </a>
            .
          </small>
        </p>
        <input
          type="submit"
          value="Create your account"
        />
      </form>
    </article>
  </div>
</div>
`;

exports[`SignupForm renders with form_options 1`] = `
<div>
  <div
    className="layout"
  >
    <header>
      <h1
        className="au-display-xl"
      >
        Let’s get started
      </h1>
    </header>
    <article
      role="main"
    >
      <form
        id="signup"
        method="post"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <input
          id="csrf_token"
          name="csrf_token"
          type="hidden"
          value="sometoken"
        />
        <div
          styleName="user-type"
        >
          <div
            className="field"
          >
            <fieldset>
              <legend>
                Have you got digital expertise to sell to government? Or do you want to buy digital products and services on behalf of your government office? Choose the option that matches your situation.
              </legend>
              <div>
                <span>
                  <input
                    checked={false}
                    disabled={false}
                    id="user_type-buyer"
                    name="user_type"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    type="radio"
                    value="buyer"
                  />
                  <label
                    htmlFor="user_type-buyer"
                  >
                    <span>
                      Buyer
                      <p>
                        I want to buy on behalf of
                        <br />
                        government.
                      </p>
                    </span>
                  </label>
                </span>
                <span>
                  <input
                    checked={false}
                    disabled={false}
                    id="user_type-seller"
                    name="user_type"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    type="radio"
                    value="seller"
                  />
                  <label
                    htmlFor="user_type-seller"
                  >
                    <span>
                      Seller
                      <p>
                        I want to sell digital products or
                        <br />
                        services.
                      </p>
                    </span>
                  </label>
                </span>
              </div>
            </fieldset>
          </div>
        </div>
        Now enter your name and your work email address.
        <div
          className="field"
        >
          <label
            className="question-heading"
            htmlFor="name"
          >
            Full name
          </label>
          <input
            className=""
            disabled={false}
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
          />
        </div>
        <div
          className="field"
        >
          <label
            className="question-heading"
            htmlFor="email_address"
          >
            Email address
          </label>
          <p
            className="hint"
            id="email_address-hint"
          >
            <br />
          </p>
          <input
            className=""
            disabled={false}
            id="email_address"
            name="email_address"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="email"
          />
        </div>
        <p>
          <small>
            By creating an account you confirm your acceptance of our
            <a
              href="/terms-of-use"
              rel="noopener noreferrer"
              target="_blank"
            >
               Terms of Use
            </a>
            .
          </small>
        </p>
        <input
          type="submit"
          value="Create your account"
        />
      </form>
    </article>
  </div>
</div>
`;
